hamster_lib/objects.py

```diff
@@ -366,9 +366,10 @@ def __init__(self, activity, start, end=None, pk=None, description=None, tags=No
         self.start = start
         self.end = end
         self.description = description
-        self.tags = set()
+        self.tags = []
         if tags:
-            self.tags = set(tags)
+            tags = set(tags)
+            self.tags = [Tag(name = x) for x in tags]
```

Comment:
Whist you are correct that this should be a set of ``Tag`` instances the mistake is already in L356! The constructor should expect an iterable of such ``Tag`` instances, *not strings*. I [created an issue about this](https://projecthamster.atlassian.net/browse/LIB-237).